src/Microsoft.ML.Data/Data/Conversion.cs

```diff
@@ -1430,14 +1402,14 @@ private bool TryParseSigned(long max, ref TX span, out long result)
             if (!TryParseNonNegative(text, ichMin, ichLim, out val))
             {
                 // Check for acceptable NA forms: ? NaN NA and N/A.
-                result = -max - 1;
+                result = null;
                 return IsStdMissing(ref span);
             }
```

Comment:
Same here. We do succeed in parsing `NA` into a missing integer any more. For floats, sure, but not integers.

Reply:
I was returning null and that would have caused an exception anyways but now I have removed the boolean return value and instead I'm just using nullable long to determine error condition,

---
In reply to: [212041270](https://github.com/dotnet/machinelearning/pull/683#discussion_r212041270) [](ancestors = 212041270)